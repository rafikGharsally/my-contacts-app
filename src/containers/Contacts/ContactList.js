import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { get } from 'axios';
import { fetchContacts } from '../../actions/Contacts';



class ContactList extends Component {

  componentDidMount() {
   this.props.fetchContacts();
  }


  render() {
    const { contacts } = this.props;
    const notFound = !contacts.data ||  contacts.data.length < 0;
    return (
      <div>
        <h2>Contact List</h2>
        { contacts.data && contacts.data.length > 0 && contacts.data.map((contact) => {
          return (
            <div key={contact.id}>{contact.name}</div>
          )
        })
        }
        { notFound &&
          <h3>no contacts to display...</h3>
        }
      </div>

    );
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ fetchContacts }, dispatch);
};


const mapStateToProps = state => {
  console.log('state---' ,state);
  return {
    contacts: state.contacts
  }
};


export default connect(
  mapStateToProps, mapDispatchToProps)(ContactList);

