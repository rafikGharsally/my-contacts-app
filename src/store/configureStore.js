import { createStore, applyMiddleware, compose } from 'redux';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
import axios from 'axios';
import axiosMiddleware from 'redux-axios-middleware';
import thunk from 'redux-thunk';
import rootReducer from '../reducers'

export default function createReduxStore() {
  // add a browser history
  const history = createHistory();

  // intercept and dispatch action here
  const middleware = routerMiddleware(history);

  const client = axios.create();

  const clientOptions = {
    returnRejectedPromiseOnError: true,
    errorSuffix: '_ERROR',
    successSuffix: '_SUCCESS'
  };

  // Add the reducer to your store on the `router` key
  // Also apply our middleware for navigating
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const store = createStore(
    rootReducer,
    // initialState
    {},
    composeEnhancers(applyMiddleware(middleware, thunk, axiosMiddleware(client, clientOptions))
    ));

  if (process.env.NODE_ENV !== "production") {
    if (module.hot) {
      module.hot.accept('../reducers', () => {
        const newRootReducer = require('../reducers').default;
        store.replaceReducer(newRootReducer);
      });
    }
  }

  return store;
};