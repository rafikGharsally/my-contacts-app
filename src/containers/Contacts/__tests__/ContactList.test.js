import React from 'react';
import ContactList  from '../ContactList';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from 'react-router-dom';

describe('Contacts', () => {
  const store = configureStore([
    thunk,
  ])();
  let fakeContacts = {
    data:[
      {
        "id": 1,
        "name": "Rafik Gharsalli",
        "job_title": "software engineer",
        "address": "1234 test test",
        "picture": "",
        "email": "rafikgharsally@gmail.com",
        "phone": "5145891865"
      }
    ]};

  it('renders without crashing', () => {
    const wrapper = shallow(
      <ContactList contacts={null} store={store} />
    ).dive();
    expect(wrapper.length).toEqual(1);
  });

  it('renders without crashing with a fake contact list', () => {
    const wrapper = shallow(
      <ContactList contacts={fakeContacts} store={store} />
    ).dive();
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('.contacts').length).toBe(1);
  });


  it('renders and matches our snapshot', () => {
    const component = renderer.create((<Router><ContactList store={store} /></Router>));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});