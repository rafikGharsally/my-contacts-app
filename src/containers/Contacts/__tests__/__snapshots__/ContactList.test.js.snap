// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts renders and matches our snapshot 1`] = `
<div
  className="contacts"
>
  <h3>
    No contacts to display...
  </h3>
</div>
`;
